src/main/java/be/seeseemelk/mockbukkit/WorldMock.java

```diff
@@ -1118,41 +1123,42 @@ else if (clazz == Firework.class)
 		{
 			return new FireworkMock(server, UUID.randomUUID());
 		}
-		else if (clazz == Hanging.class)
-		{
-			// LeashHitch has no direction and is always centered
-			if (LeashHitch.class.isAssignableFrom(clazz))
-			{
-				throw new UnimplementedOperationException();
-			}
-			BlockFace spawnFace = BlockFace.SELF;
-			BlockFace[] faces = (ItemFrame.class.isAssignableFrom(clazz))
-					? new BlockFace[]{ BlockFace.NORTH, BlockFace.EAST, BlockFace.SOUTH, BlockFace.WEST, BlockFace.UP, BlockFace.DOWN }
-					: new BlockFace[]{ BlockFace.NORTH, BlockFace.EAST, BlockFace.SOUTH, BlockFace.WEST };
-			for (BlockFace face : faces)
-			{
-				Block block = this.getBlockAt(location.add(face.getModX(), face.getModY(), face.getModZ()));
-				if (!block.getType().isSolid() && (block.getType() != Material.REPEATER && block.getType() != Material.COMPARATOR))
-					continue;
-
-				boolean taken = false;
-
-				// TODO: Check if the entity's bounding box collides with any other hanging entities.
-
-				if (taken)
-					continue;
-
-				spawnFace = face;
-				break;
-			}
-			if (spawnFace == BlockFace.SELF)
-			{
-				spawnFace = BlockFace.SOUTH;
-			}
-			spawnFace = spawnFace.getOppositeFace();
-			// TODO: Spawn entities here.
-			throw new UnimplementedOperationException();
-		}
+// 		TODO: This code is incomplete and needs to be worked on
+//		else if (clazz == Hanging.class)
+//		{
+//			// LeashHitch has no direction and is always centered
+//			if (LeashHitch.class.isAssignableFrom(clazz))
+//			{
+//				throw new UnimplementedOperationException();
+//			}
+//			BlockFace spawnFace = BlockFace.SELF;
+//			BlockFace[] faces = (ItemFrame.class.isAssignableFrom(clazz))
+//					? new BlockFace[]{ BlockFace.NORTH, BlockFace.EAST, BlockFace.SOUTH, BlockFace.WEST, BlockFace.UP, BlockFace.DOWN }
+//					: new BlockFace[]{ BlockFace.NORTH, BlockFace.EAST, BlockFace.SOUTH, BlockFace.WEST };
+//			for (BlockFace face : faces)
+//			{
+//				Block block = this.getBlockAt(location.add(face.getModX(), face.getModY(), face.getModZ()));
+//				if (!block.getType().isSolid() && (block.getType() != Material.REPEATER && block.getType() != Material.COMPARATOR))
+//					continue;
+//
+//				boolean taken = false;
+//
+//				// TODO: Check if the entity's bounding box collides with any other hanging entities.
+//
+//				if (taken)
+//					continue;
+//
+//				spawnFace = face;
+//				break;
+//			}
+//			if (spawnFace == BlockFace.SELF)
+//			{
+//				spawnFace = BlockFace.SOUTH;
+//			}
+//			spawnFace = spawnFace.getOppositeFace();
+//			// TODO: Spawn entities here.
+//			throw new UnimplementedOperationException();
+//		}
```

Comment:
This code was the only part that depended on the location directly, and it seems incomplete. I only see UnimplementedOperationException being trown. I commented it out, since it's important code but that was not finished yet

Reply:
Generally commenting out code is a bad practice. I'm not quite sure why, It might be because it gets harder to read the code that is actually doing things. (I will look into this now)

Reply:
Seems like paper has moved over to have a full data-based implementation now. When I look at it, this method is about 450 lines of if statements, this should be fixed for our own sake. I will think about how to do this. But that's a separate pr